src/pages/task/taskTabs/buildBaronAndAnnotations/useBuildBaronVariables.ts

```diff
@@ -0,0 +1,48 @@
+import { useQuery } from "@apollo/client";
+import {
+  GetBuildBaronConfiguredQuery,
+  GetBuildBaronConfiguredQueryVariables,
+} from "gql/generated/types";
+import { GET_BUILD_BARON_CONFIGURED } from "gql/queries";
+import { statuses } from "utils";
+
+const { isFailedTaskStatus } = statuses;
+interface UseBuildBaronVariablesType {
+  task: {
+    id: string;
+    execution: number;
+    status: string;
+  };
+  hasAnnotation: boolean;
+  canModifyAnnotation: boolean;
+}
+const useBuildBaronVariables = ({
+  task,
+  hasAnnotation,
+  canModifyAnnotation,
```

Comment:
all of these fields are part of the task gql type. does it makes sense to expand or consolidate these fields under one term?